controllers/datadogagent/utils_kubelet.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package datadogagent
+
+import (
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/api/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func getKubeletEnvVars(dda *datadoghqv1alpha1.DatadogAgent) []corev1.EnvVar {
+	kubeletVars := make([]corev1.EnvVar, 0)
+
+	// Host valueFrom
+	var kubeletHostValueFrom *corev1.EnvVarSource
+	if dda.Spec.Agent.Config.Kubelet != nil && dda.Spec.Agent.Config.Kubelet.Host != nil {
+		kubeletHostValueFrom = dda.Spec.Agent.Config.Kubelet.Host
+	} else {
+		kubeletHostValueFrom = &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: FieldPathStatusHostIP,
+			},
+		}
+	}
+
+	kubeletVars = append(kubeletVars, corev1.EnvVar{
+		Name:      datadoghqv1alpha1.DDKubeletHost,
+		ValueFrom: kubeletHostValueFrom,
+	})
+
+	// TLS Verify
+	if dda.Spec.Agent.Config.Kubelet != nil && dda.Spec.Agent.Config.Kubelet.TLSVerify != nil {
+		kubeletVars = append(kubeletVars, corev1.EnvVar{
+			Name:  datadoghqv1alpha1.DDKubeletTLSVerify,
+			Value: datadoghqv1alpha1.BoolToString(dda.Spec.Agent.Config.Kubelet.TLSVerify),
+		})
+	}
+
+	// CA Path
+	if dda.Spec.Agent.Config.Kubelet != nil && (dda.Spec.Agent.Config.Kubelet.AgentCAPath != "" || dda.Spec.Agent.Config.Kubelet.HostCAPath != "") {
+		kubeletVars = append(kubeletVars, corev1.EnvVar{
+			Name:  datadoghqv1alpha1.DDKubeletCAPath,
+			Value: getAgentCAPath(dda),
+		})
+	}
+
+	return kubeletVars
+}
+
+func getKubeletVolumes(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Volume {
```

Comment:
could you modify the interface in order to use `datadoghqv1alpha1.DatadogAgentSpec` instead. I don't think you need to access the Object metadata  or the status.
It will help to implement the CRD defaulting improvement task.

Reply:
Okay so I aligned everything to `dda` instead :D, I'll go back and align back to `Spec` then